### PR TITLE
feat: Create rich-link element 

### PR DESCRIPTION
**Pull request by dskamiotis**
## What does this change?
This PR adds a `rich-link` element replicating the existing element in Composer PROD. 

This PR also:
- adds "rich-link" as a string as a defaultElementTransform 
- creates demo rich-link element for sandbox with button, generating example text for link to article
- add `min-height` styling to ElementWrapper to ensure grey body gives enough space to arrow/controls to be shown clearly

### Images - Element in Composer CODE (tested in flexible-content)
Screenshot 2021-08-06 at 16 05 00 

## How to test in flexible-content

Run `yarn start` locally and add an rich-link element.

To test this with `flexible-content` locally, I used [`yalc`](https://github.com/wclr/yalc). If you want to do the same: 
1. Install `yalc` globally with `npm i yalc -g` or `yarn global add yalc`
2. Run `yarn build` in your local `prosemirror-elements` from this branch
3. Run `yalc publish` in the same directory
4. Run `yalc add @guardian/prosemirror-elements` from the `composer` directory of your local `flexible-content`

When pasting the article in composer ensure the article is listed in `/live/` CODE ensure the article is available as part of https://capi-proxy.local.dev-gutools.co.uk/live/
## Accessibility

- [ ] [Tested with screen reader](https://github.com/guardian/source/blob/main/docs/06-accessibility.md#screen-readers)
- [ ] [Navigable with keyboard](https://github.com/guardian/source/blob/main/docs/06-accessibility.md#keyboard-navigation)
- [ ] [Colour contrast passed](https://github.com/guardian/source/blob/main/docs/06-accessibility.md#colour-contrast)
- [ ] [The change doesn't use only colour to convey meaning](https://github.com/guardian/source/blob/main/docs/06-accessibility.md#use-of-colour)
- [ ] [Interactive elements show a focus ring when focused by keyboard](https://github.com/guardian/accessibility/blob/main/editorial-tools-guidelines.md#appendix-the-focus-ring)
- [ ] [Semantic elements have been used where possible](https://github.com/guardian/accessibility/blob/main/editorial-tools-guidelines.md#appendix-semantic-html)

### CODE REVIEW

File: src/renderers/react/ElementWrapper.tsx [hunk 1]
```diff
@@ -32,6 +32,7 @@ const Body = styled("div")`
       opacity: 1;
     }
   }
+  min-height: 134px;
```
Comment (jonathonherbert) [inline src/renderers/react/ElementWrapper.tsx:35]: It'd be nice to avoid hardcoding this – but that'll need a refactor that probably belongs in a separate PR, so we're fine for now!

File: demo/index.ts [hunk 2]
```diff
@@ -180,6 +184,14 @@ const createEditor = (server: CollabServer) => {
     })
   );
 
+  editorElement.appendChild(
+    createElementButton("Add demo rich-link element", richlinkElementName, {
```
Comment (jonathonherbert) [inline demo/index.ts:188]:
```suggestion
    createElementButton("Add rich-link element", richlinkElementName, {
```

No demo, this! 😁

File: src/elements/rich-link/RichlinkForm.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import { CustomDropdownView } from "../../renderers/react/customFieldViewComponents/CustomDropdownView";
+import type { richlinkFields } from "./RichlinkSpec";
+
+type Props = {
+  fieldValues: FieldNameToValueMap<typeof richlinkFields>;
+  errors: FieldValidationErrors;
+  fields: FieldNameToField<typeof richlinkFields>;
+};
+
+export const RichlinkElementTestId = "RichlinkElement";
```
Comment (jonathonherbert) [inline src/elements/rich-link/RichlinkForm.tsx:15]: I can't see that we use this element in our integration tests – if we don't test this element anywhere, we shouldn't need property.

**Review by jonathonherbert — COMMENTED** [review]
Tested in Composer, looks great! A few minor comments and we should be good to go.

File: src/renderers/react/ElementWrapper.tsx (hunk 1 above)
Comment (dskamiotis) [reply src/renderers/react/ElementWrapper.tsx:35]: 👍

File: src/elements/rich-link/RichlinkForm.tsx (hunk 3 above)
Comment (dskamiotis) [reply src/elements/rich-link/RichlinkForm.tsx]: @jonathonherbert thanks well spotted, perhaps another PR to address adding an integration test for this element would be appropriate.

Comment (jonathonherbert) [reply src/elements/rich-link/RichlinkForm.tsx]: Yes, let's not worry about this in this PR 👍 We probably need a better a story for integration testing our elements, which at the moment would be quite painful.

**Review by jonathonherbert — APPROVED** [review]
Changes look good! 👍
